### PR TITLE
Update Rewards to Align with Spec 

### LINKED ISSUE #593: Implement Deterministic Sqrt Helper Function
The Eth2.0 implementation teams have decided it's best off to define a deterministic sqrt helper function rather than the built in implementation for various languages. This is to reduce errors across implementations on consensus critical calculations which will be used by the rewards scheme. 

To resolve this issue, we should implement the helper function mentioned [here](https://github.com/ethereum/eth2.0-specs/pull/13) and use it in our casper package.

### PR DESCRIPTION
**Pull request by nisdas**
This resolves #593 by adding a helper function to get the integer root of a number and also makes changes to calculation of rewards by removing any floating point arithmetic

### CODE REVIEW

File: beacon-chain/casper/incentives.go [hunk 1]
```diff
@@ -36,8 +34,8 @@ func CalculateRewards(
 				if voterIndex == validatorIndex {
 					voted = true
 					balance := validators[validatorIndex].GetBalance()
-					newbalance := uint64(balance + (balance/rewardQuotient)*depositFactor)
-					validators[validatorIndex].Balance = newbalance
+					newbalance := int64(balance) + int64(balance/rewardQuotient)*(2*int64(totalParticipatedDeposit)-int64(totalDeposit))/int64(totalDeposit)
```
Comment (prestonvanloon) [inline beacon-chain/casper/incentives.go]: Why conversion to int64 rather than uint64?

File: shared/math_helper_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+			number: 34989843,
+			root:   5915,
+		},
+		{
+			number: 97282,
+			root:   311,
+		},
+	}
+
+	for _, testVals := range tt {
```
Comment (prestonvanloon) [inline shared/math_helper_test.go]: Nice table driven test :)

Comment (djrtwo) [reply shared/math_helper_test.go]: seconded.

File: beacon-chain/casper/incentives.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/casper/incentives.go]: so according to the spec ```balance/rewardQuotient*(2*totalParticipatedDeposit-totalDeposit/totalDeposit``` can also be negative so casting to `int64` is required over here

**Pull request merged**